### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by sbernhard**

### CODE REVIEW

**Comment by sbernhard** [comment]
[test foreman_salt]

File: app/services/foreman_salt/fact_parser.rb [hunk 1]
```diff
@@ -56,14 +56,15 @@ def interfaces
 
       facts.each do |fact, value|
         next unless value && fact.to_s =~ /^ip_interfaces/
-        (_, interface_name, _) = fact.split(FactName::SEPARATOR)
+
+        (_, interface_name,) = fact.split(FactName::SEPARATOR)
```
Comment (ekohl) [inline app/services/foreman_salt/fact_parser.rb]:
```suggestion
        (_, interface_name) = fact.split(FactName::SEPARATOR)
```

File: app/models/foreman_salt/concerns/hostgroup_extensions.rb [hunk 2]
```diff
@@ -28,31 +28,29 @@ def inherited_salt_module_ids
       end
 
       def salt_proxy
-        return super unless ancestry.present?
-        SmartProxy.find_by_id(inherited_salt_proxy_id)
-      end
-
-      def inherited_salt_proxy_id
         if ancestry.present?
```
Comment (ekohl) [inline app/models/foreman_salt/concerns/hostgroup_extensions.rb]: Reading this suggests it reverses the order. However, the `inherited_salt_proxy_id` prefers the actual ID which is a bit confusing to me. Perhaps you should use `real_salt_proxy_id` or `actual_salt_proxy_id`.

File: app/models/foreman_salt/concerns/hostgroup_extensions.rb [hunk 3]
```diff
@@ -28,31 +28,29 @@ def inherited_salt_module_ids
...
-          self[:salt_proxy_id] || self.class.sort_by_ancestry(ancestors.where('salt_proxy_id is not NULL')).last.try(:salt_proxy_id)
+          SmartProxy.with_features('Salt').find_by_id(inherited_salt_proxy_id)
         else
-          salt_proxy_id
+          super
```
Comment (ekohl) [inline app/models/foreman_salt/concerns/hostgroup_extensions.rb]: What does the super do? Given you already search exactly for the proxy with the right ID, this feels redundant.

File: app/models/foreman_salt/concerns/hostgroup_extensions.rb (hunk 2 above)
Comment (sbernhard) [reply app/models/foreman_salt/concerns/hostgroup_extensions.rb]: same behavior is in foreman_chef. Therefore I would stick with this for now.

File: app/models/foreman_salt/concerns/hostgroup_extensions.rb (hunk 3 above)
Comment (sbernhard) [reply app/models/foreman_salt/concerns/hostgroup_extensions.rb]: same behavior is in foreman_chef. Therefore I would stick with this for now.

**Pull request merged**